crates/ide-completion/src/completions/dot.rs

```diff
@@ -1095,4 +1095,81 @@ fn test(s: S<Unknown>) {
             "#]],
         );
     }
+
+    #[test]
+    fn assoc_impl_1() {
```

Comment:
it would be good to at least additionally have a test for this in `hir-ty` as well.